### PR TITLE
Update AzureCoreToolsMissingNupkgInstaller to work with 2021.1

### PR DESCRIPTION
**Pull request by maartenba**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/AzureCoreToolsMissingNupkgInstaller.kt [hunk 1]
```diff
@@ -81,10 +90,16 @@ class AzureCoreToolsMissingNupkgInstaller : StartupActivity {
...
-                            hasKnownFileSuffix(file) && // Only process certain file types
-                            event.requestor !is FileDocumentManagerImpl && // Don't process when typing in document
-                            file.length > 0) {
+                    if (hasKnownFileSuffix(file) &&
+                            existsInCurrentProject(projectFileIndex, file) &&
+                            isNewOrMarkedForProcessing(file, event)) {
+
+                        // First pass(es), no content will be in the file.
+                        // If that is the case, mark the file for processing on later changes.
+                        if (file.length < 25 || event is VFileCreateEvent) {
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/AzureCoreToolsMissingNupkgInstaller.kt:99]: Why do we check for `length < 25` here, not just `!= 0`?

**Review by sdubov — APPROVED** [review]
I'm concerned a bit about NuGet installation. We install NuGets without asking a user about it. We do this on every vfs  change + we have this option enabled by default. What do you think about making it as a quick fix instead?

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/AzureCoreToolsMissingNupkgInstaller.kt (hunk 1 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/AzureCoreToolsMissingNupkgInstaller.kt:99]: When creating a new file from our templates, we get these VFS events:
* Create - length == 0
* Update - length == 3 (have seen 10 as well, chose 25)
* Update - length == actual template length

**Comment by maartenba** [comment]
We have had this since a long time, this PR is just fixing it for 2021.1.
https://blog.jetbrains.com/dotnet/2019/05/09/building-azure-functions-sql-database-improvements-azure-toolkit-rider-2019-1/

Happy to re-evaluate or make it optional.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/AzureCoreToolsMissingNupkgInstaller.kt (hunk 1 above)
Comment (sdubov) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/AzureCoreToolsMissingNupkgInstaller.kt:99]: I would add some explanation in code :)

**Comment by maartenba** [comment]
As discussed offline, merging as-is and moving to a new Rider extension point later on when it is available.

**Comment by sdubov** [comment]
Created a related ticket to add an extension point for `RiderNewFileFromTemplateActionBase` [RIDER-57928](https://youtrack.jetbrains.com/issue/RIDER-57928).

**Pull request merged**